prow/tide/tide.go

```diff
@@ -834,75 +825,102 @@ func (c *Controller) mergePRs(sp subpool, prs []PullRequest) error {
+						// overzealous branch protection setting will not allow the robot to
+						// push to a specific branch.
+						// We won't be able to merge the other PRs.
+						return false, fmt.Errorf("branch needs to be configured to allow this robot to push: %v", err)
+					} else if _, ok = err.(github.MergeCommitsForbiddenError); ok {
+						// Github let us know that the merge method configured for this repo
+						// is not allowed by other repo settings, so we should let the admins
+						// know that the configuration needs to be updated.
+						// We won't be able to merge the other PRs.
+						return false, fmt.Errorf("Tide needs to be configured to use the 'rebase' merge method for this repo or the repo needs to allow merge commits: %v", err)
+					} else if _, ok = err.(github.UnmergablePRError); ok {
+						return true, fmt.Errorf("PR is unmergable. Do the Tide merge requirements match the GitHub settings for the repo? %v", err)
+					} else {
+						return true, err
 					}
-				} else if _, ok = err.(github.UnauthorizedToPushError); ok {
-					// Github let us know that the token used cannot push to the branch.
-					// Even if the robot is set up to have write access to the repo, an
-					// overzealous branch protection setting will not allow the robot to
-					// push to a specific branch.
-					log.WithError(err).Error("Merge failed: Branch needs to be configured to allow this robot to push.")
-					// We won't be able to merge the other PRs.
-					return augmentError(err, pr)
-				} else if _, ok = err.(github.MergeCommitsForbiddenError); ok {
-					// Github let us know that the merge method configured for this repo
-					// is not allowed by other repo settings, so we should let the admins
-					// know that the configuration needs to be updated.
-					log.WithError(err).Error("Merge failed: Tide needs to be configured to use the 'rebase' merge method for this repo or the repo needs to allow merge commits.")
-					// We won't be able to merge the other PRs.
-					return augmentError(err, pr)
-				} else if _, ok = err.(github.UnmergablePRError); ok {
-					log.WithError(err).Error("Merge failed: PR is unmergable. Do the Tide merge requirements match the GitHub settings for the repo?")
-					break
 				} else {
-					log.WithError(err).Error("Merge failed.")
-					return augmentError(err, pr)
+					// Successful merge!
+					return true, nil
 				}
-			} else {
-				log.Info("Merged.")
-				merged = append(merged, int(pr.Number))
-				// If we have more PRs to merge, sleep to give Github time to recalculate
-				// mergeability.
-				if i+1 < len(prs) {
-					time.Sleep(time.Second * 3)
-				}
-				break
 			}
+			// We ran out of retries. Return the last transient error.
+			return true, err
+		}
+
+		keepTrying, err := mergePR()
+		if err != nil {
+			log.WithError(err).Error("Merge failed.")
+			errs = append(errs, err)
+			failed = append(failed, int(pr.Number))
```

Comment:
I think you want a `continue` at the end of this block?

Reply:
Or an `else` for the block where you add this number to the merged list, so in failure case you still do the `keepTrying` check

Reply:
Done.